zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +150,238 @@ def compute(self, today, assets, out, close, volume):
+    were calculated, take for example the .15 in MSFT's column. This is the
+    correlation coefficient between SPY's returns looking back from 2017-03-17
+    (-.03, -.02, -.01, 0, .01) and MSFT's returns (.03, -.03, .02, -.02, .04).
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingReturnsRegression`
+    """
+    params = ['baseline_asset']
+
+    def __new__(cls,
+                baseline_asset,
+                returns_length,
+                correlation_length,
+                mask=NotSpecified,
+                **kwargs):
+        if mask is not NotSpecified:
+            # Make sure we do not filter out the asset of interest.
+            mask = mask | SingleAsset(asset=baseline_asset)
+        return super(RollingPearsonOfReturns, cls).__new__(
+            cls,
+            baseline_asset=baseline_asset,
+            inputs=[Returns(window_length=returns_length)],
+            window_length=correlation_length,
+            mask=mask,
+            **kwargs
+        )
+
+    def compute(self, today, assets, out, data, baseline_asset):
+        asset_col = searchsorted(list(assets), baseline_asset.sid)
+        out[:] = corrcoef(data, rowvar=0)[asset_col]
+
+
+class RollingSpearmanOfReturns(RollingPearsonOfReturns):
+    """
+    Calculates the Spearman rank correlation coefficient of the returns of the
+    given asset with the returns of all other assets.
+
+    **Default Inputs:** [Returns]
+
+    Parameters
+    ----------
+    baseline_asset : zipline.assets.Asset
+        The asset to correlate with all other assets.
+    returns_length : int > 1
+        Length of the lookback window over which to compute returns.
+    correlation_length : int > 1
+        Length of the lookback window over which to compute each correlation
+        coefficient.
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingSpearmanOfReturns`
+    """
+    def compute(self, today, assets, out, data, baseline_asset):
+        asset_col = searchsorted(list(assets), baseline_asset.sid)
+        out[:] = spearmanr(data)[0][asset_col]
+
+
+class RollingRegressionOfReturns(SingleInputMixin, CustomFactor):
+    """
+    Perform an ordinary least-squares regression predicting the returns of all
+    other assets on the given asset.
+
+    **Default Inputs:** [Returns]
+
+    Parameters
+    ----------
+    baseline_asset : zipline.assets.Asset
+        The asset to regress against all other assets.
+    returns_length : int > 1
+        Length of the lookback window over which to compute returns.
+    regression_length : int > 1
+        Length of the lookback window over which to compute each regression.
+
+    Note
+    ----
+    This factor is designed to return five outputs:
+        - alpha, a factor that computes the intercepts of the regressions.
+        - beta, a factor that computes the slopes of the regressions.
+        - r_value, the correlation coefficient
```

Comment:
The latter three values should probably still make clear that they're for the regression between each asset and the target.